crucible-llvm/src/Lang/Crucible/LLVM/SimpleLoopFixpoint.hs

```diff
@@ -0,0 +1,748 @@
+        (\entry -> fromJust $ MapF.lookup entry reverse_substitution)
+        substitution
+      normal_substitution = MapF.filterWithKey
+        (\variable _entry ->
+          Just Refl == W4.testEquality variable (fromJust $ MapF.lookup variable uninterpreted_constant_substitution))
+        substitution
+  in
+  (normal_substitution, uninterpreted_constant_substitution)
+
+
+userSymbol' :: String -> W4.SolverSymbol
+userSymbol' = fromRight (C.panic "SimpleLoopFixpoint.userSymbol'" []) . W4.userSymbol
+
+
+simpleLoopFixpoint ::
+  forall sym ext p rtp blocks init ret .
+  (C.IsSymInterface sym, C.HasLLVMAnn sym, ?memOpts :: C.MemOptions) =>
+  sym ->
+  C.CFG ext blocks init ret ->
+  C.GlobalVar C.Mem ->
+  (MapF (W4.SymExpr sym) (FixpointEntry sym) -> W4.Pred sym -> IO (MapF (W4.SymExpr sym) (W4.SymExpr sym), W4.Pred sym)) ->
+  IO (C.ExecutionFeature p sym ext rtp)
+simpleLoopFixpoint sym cfg@C.CFG{..} mem_var fixpoint_func = do
+  let ?ptrWidth = knownNat @64
+
+  verbSetting <- W4.getOptionSetting W4.verbosity $ W4.getConfiguration sym
+  verb <- fromInteger <$> W4.getOpt verbSetting
+
+  let flattenWTOComponent = \case
+        C.SCC C.SCCData{..} ->  wtoHead : concatMap flattenWTOComponent wtoComps
+        C.Vertex v -> [v]
+  let loop_map = Map.fromList $ mapMaybe
+        (\case
+          C.SCC C.SCCData{..} -> Just (wtoHead, wtoHead : concatMap flattenWTOComponent wtoComps)
+          C.Vertex{} -> Nothing)
+        (C.cfgWeakTopologicalOrdering cfg)
+
+  fixpoint_stuff_ref <- newIORef @(FixpointStuff sym blocks) $ FixpointStuff
+    { fixpointStatus = BeforeFixpoint
+    , fixpointBlockId = undefined
+    , fixpointAssumptionFrameIdentifier = undefined
+    , fixpointSubstitution = undefined
+    , fixpointRegMap = undefined
+    , fixpointMemSubstitution = undefined
+    , fixpointLoopCondition = undefined
+    , fixpointLoopIndexBound = undefined
+    }
+
+  return $ C.ExecutionFeature $ \exec_state -> do
+    let ?logMessage = \msg -> when (verb >= (3 :: Natural)) $ do
+          let h = C.printHandle $ C.execStateContext exec_state
+          System.IO.hPutStrLn h msg
+          System.IO.hFlush h
+    fixpoint_stuff <- readIORef fixpoint_stuff_ref
+    case exec_state of
+      C.RunningState (C.RunBlockStart block_id) sim_state
+        | C.SomeHandle cfgHandle == C.frameHandle (sim_state ^. C.stateCrucibleFrame)
+        , Just Refl <- W4.testEquality
+            (fmapFC C.blockInputs cfgBlockMap)
+            (fmapFC C.blockInputs $ C.frameBlockMap $ sim_state ^. C.stateCrucibleFrame)
+        , Map.member (C.Some block_id) loop_map -> case fixpointStatus fixpoint_stuff of
+
+          BeforeFixpoint -> do
+            ?logMessage $ "SimpleLoopFixpoint: RunningState: " ++ show BeforeFixpoint ++ " -> " ++ show ComputeFixpoint
+            assumption_frame_identifier <- C.pushAssumptionFrame sym
+            let mem_impl = fromJust $ C.lookupGlobal mem_var (sim_state ^. C.stateGlobals)
```

Comment:
Can we make this a panic if the variable is missing? At a minimum, a `HasCallStack` constraint would make this easier to track down if messed up